### PR TITLE
Script for uploading gce template/image

### PR DESCRIPTION
**Pull request by lehinevych**

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 1a5c59811d73f8ef8f8a5a8754bfa4cba9d35f99:

`scripts/template_upload_gce.py`:
- :grey_exclamation: Line 80:5: **T001** *print statement found.*
- :grey_exclamation: Line 80:25: **S001** *found %s formatter*
- :grey_exclamation: Line 92:20: **S001** *found % formatter*
- :grey_exclamation: Line 94:9: **T001** *print statement found.*

Please, rectify these issues :smirk: .
*CFME QE Bot*

File: scripts/template_upload_gce.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+import urllib2
+from threading import Lock, Thread
+
+from utils.conf import cfme_data
+from utils.conf import credentials
+# TODO replace by mgmt system lib
+from utils.mgmt_system import GoogleCloudSystem
+from utils.providers import list_providers
+
+lock = Lock()
```
Comment (RonnyPfannschmidt) [inline scripts/template_upload_gce.py]: is this used anywhere?

**Comment by dajoRH** [comment]
Lint report for commit 3457bf8163a57bb22fa2466658f9edaa717f610e:

`scripts/template_upload_gce.py`:
- :grey_exclamation: Line 92:20: **S001** *found % formatter*

Please, rectify these issues :smirk: .
*CFME QE Bot*

File: scripts/template_upload_gce.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        gce_credentials = credentials[mgmt_sys['credentials']]
+
+        service_account = gce_credentials['service_account']
+        project = mgmt_sys['project']
+        zone = mgmt_sys['zone']
+        thread = Thread(target=upload_template,
+                        args=(project, zone, service_account, kwargs.get('image_url'),
+                            kwargs.get('template_name'), kwargs.get('bucket_name'), provider))
+        thread.daemon = True
+        thread_queue.append(thread)
```
Comment (RonnyPfannschmidt) [inline scripts/template_upload_gce.py]: is it ok to just start as many threads as there are gce providers, (i suppose we dont have that many in a call)?

Comment (psav) [inline scripts/template_upload_gce.py]: I think this is probably copying behaviour from the other uploaders

**Comment by dajoRH** [comment]
Lint report for commit 7f58e358a9146ac1b6c7b58e715879781c2cdfce:

`scripts/template_upload_gce.py`:
- :grey_exclamation: Line 91:20: **S001** *found % formatter*
- :grey_exclamation: Line 93:9: **T001** *print statement found.*

Please, rectify these issues :smirk: .
*CFME QE Bot*

File: scripts/template_upload_gce.py (hunk 1 above)
Comment (lehinevych) [inline scripts/template_upload_gce.py]: The lock is present in every template_upload but not used in scripts, so I assumed it is used externally.

File: scripts/template_upload_gce.py (hunk 2 above)
Comment (lehinevych) [inline scripts/template_upload_gce.py]: Currently we have only one and threads were added to have similar behavior as other uploaders

**Comment by dajoRH** [comment]
Lint report for commit 6c9a4ccdca8da4e1a2707e5854222461210fa895:

Everything seems all right :smile: .
*CFME QE Bot*

**Pull request merged**